File: src/main/java/com/partymakers/shareparty/domain/party/usecase/impl/CreatePartyRoomImpl.java
package com.partymakers.shareparty.domain.party.usecase.impl;

import com.partymakers.shareparty.domain.party.entity.PartyRoom;
import com.partymakers.shareparty.domain.party.port.PartyRoomRepository;
import com.partymakers.shareparty.domain.party.usecase.CreatePartyRoom;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class CreatePartyRoomImpl implements CreatePartyRoom {

    private final PartyRoomRepository repository;

    @Override
    public long createPartyRoom(String description) {
           return repository.save(new PartyRoom(description)).getDescription().getId();
    }
}

File: src/main/kotlin/com/partymakers/shareparty/domain/party/usecase/impl/CreatePartyRoomImpl.kt
package com.partymakers.shareparty.domain.party.usecase.impl

import com.partymakers.shareparty.domain.party.entity.PartyRoom
import com.partymakers.shareparty.domain.party.port.PartyRoomRepository
import com.partymakers.shareparty.domain.party.usecase.CreatePartyRoom
import org.springframework.stereotype.Repository
import org.springframework.stereotype.Service

@Service
class CreatePartyRoomImpl(
    private val repository: PartyRoomRepository
) : CreatePartyRoom {

    override fun createPartyRoom(description: String): Long =
        repository.save(PartyRoom(description)).description.id
} 